clang/lib/CodeGen/CGExprComplex.cpp

```diff
@@ -283,9 +283,47 @@ class ComplexExprEmitter
   ComplexPairTy EmitComplexBinOpLibCall(StringRef LibCallName,
                                         const BinOpInfo &Op);
 
-  QualType getPromotionType(QualType Ty) {
+  QualType HigherPrecisionTypeForComplexArithmetic(QualType ElementType,
+                                                   bool IsDivOpCode) {
+    const TargetInfo &TI = CGF.getContext().getTargetInfo();
+    const LangOptions Opts = CGF.getLangOpts();
+    if (const auto *BT = dyn_cast<BuiltinType>(ElementType)) {
+      switch (BT->getKind()) {
+      case BuiltinType::Kind::Float16: {
+        if (TI.hasFloat16Type() && !TI.hasLegalHalfType())
+          return CGF.getContext().getComplexType(CGF.getContext().FloatTy);
+        break;
+      }
+      case BuiltinType::Kind::BFloat16: {
+        if (TI.hasBFloat16Type() && !TI.hasFullBFloat16Type())
+          return CGF.getContext().getComplexType(CGF.getContext().FloatTy);
+        break;
+      }
+      case BuiltinType::Kind::Float:
+        return CGF.getContext().getComplexType(CGF.getContext().DoubleTy);
+        break;
+      case BuiltinType::Kind::Double: {
+        if (TI.hasLongDoubleType())
+          return CGF.getContext().getComplexType(CGF.getContext().LongDoubleTy);
+       else
+          return CGF.getContext().getComplexType(CGF.getContext().DoubleTy);
```

Comment:
```suggestion
        return CGF.getContext().getComplexType(CGF.getContext().DoubleTy);
```
No `else` after a `return`.